manager/snap.go

```diff
@@ -79,6 +87,126 @@ func (snap *snap) GetProxySettings() (proxy.Settings, error) {
 	return res, nil
 }
 
+// ConfigureStoreProxyFromURL downloads an assertion list from the snap store
+// proxy located at proxyURL and sets up snapd to use the proxy and store ID
+// obtained from the retrieved assertions.
+//
+// If the local snap store proxy instance is operating in an air-gapped
+// environment, downloading the assertion list from the proxy will not be
+// possible and an appropriate error will be returned. For such use-cases, the
+// store proxy must be configured via a call to ConfigureStoreProxy instead.
+//
+// If snap also needs to use HTTP/HTTPS proxies to talk to the outside world,
+// these need to be configured separately before invoking this method via a
+// call to SetProxy.
+func (snap *Snap) ConfigureStoreProxyFromURL(proxyURL string) error {
+	u, err := url.Parse(proxyURL)
+	if err != nil {
+		return errors.Annotate(err, "proxy URL not valid")
+	}
+	if u.Scheme != "http" && u.Scheme != "https" {
+		return errors.NotValidf("proxy URL scheme %q", u.Scheme)
+	}
+
+	res, err := http.Get(proxyURL + "/v2/auth/store/assertions")
```

Comment:
Considering you've parsed the `proxyURL` can you not ensure that you're adding `u.Schema + u.User + u.Host`?

Reply:
We probably also need the port in there. But I guess path.Join would be safer here. I should probably also update the findStoreID code (if we don't extract it out of this PR) to be tolerant of trailing slashes.

Reply:
Also, I just realized that I need to redact/remove user creds when reporting errors!